src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -190,6 +206,34 @@ private void updateNeighborhoodReferencingPath(IModel model, FormulaUsablePath f
          formulaUsablePath.ReplaceWith(neighborhoodPath);
       }
 
+      public void ExpandLumenSegmentReferencesIn(IModel model)
+      {
+         model.Root.GetAllChildren<IUsingFormula>(x => x.Formula.IsReferencingLumenSegment())
+            .Each(x => x.Formula.ObjectPaths.Each(path => updateLumenSegmentReferencingPath(path, x)));
+      }
+
+      private void updateLumenSegmentReferencingPath(FormulaUsablePath formulaUsablePath, IUsingFormula usingFormula)
+      {
+         var pathAsList = formulaUsablePath.ToList();
+         var firstIndex = pathAsList.FindIndex(x => x == LUMEN_SEGMENT);
```

Comment:
it would be very easy to add a MUCOSA_SEGMENT if needed

Reply:
cool! :)